include/nbl/builtin/glsl/math/complex.glsl

```diff
@@ -0,0 +1,35 @@
+// Copyright (C) 2018-2020 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_MATH_COMPLEX_INCLUDED_
+#define _NBL_MATH_COMPLEX_INCLUDED_
+
+#include <nbl/builtin/glsl/math/constants.glsl>
+#include <nbl/builtin/glsl/math/functions.glsl>
+
+vec2 nbl_glsl_eITheta(in float _theta)
+{
+    // Use sincos from math/functions.glsl?
+    float r = cos(_theta);
+    float i = sin(_theta);
+    return vec2(r, i);
+}
+
+vec2 nbl_glsl_complex_mul(in vec2 rhs, in vec2 lhs)
+{
+    float r = rhs.x * lhs.x - rhs.y * lhs.y;
+    float i = rhs.x * lhs.y + rhs.y * lhs.x;
+    return vec2(r, i);
+}
+
+vec2 nbl_glsl_complex_add(in vec2 rhs, in vec2 lhs)
+{
+    return rhs + lhs;
+}
+
+vec2 nbl_glsl_complex_conjugate(in vec2 complex) {
```

Comment:
nice functions... although instead of a `vec2` type I'd define a 
```glsl
#define nbl_glsl_complex vec2
#define nbl_glsl_cvec2 mat2
#define nbl_glsl_cvec3 mat3x2
#define nbl_glsl_cvec4 mat4x2
```